core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -673,7 +673,7 @@ private[spark] object Utils
       throw new IOException(s"Failed to create directory ${targetDir.getPath}")
     }
     val dest = new File(targetDir, filename.getOrElse(path.getName))
-    if (fs.isFile(path)) {
+    if (fs.getFileStatus(path).isFile) {
```

Comment:
Although the semantics of this sentence are different from the original, our function maintains the same semantics.
In the original logic, if `path` does not exist, it will go to the `else statement`, and calling `fs.listStatus(path)` will also generate an exception: `FileNotFoundException`.